sources/editor/Xenko.Assets.Presentation/Templates/Core/ProjectExecutable.Linux/$ProjectName$.csproj.t4

```diff
@@ -21,4 +24,12 @@
     <PackageReference Include="Xenko" Version="<#= Xenko.Assets.XenkoConfig.GetLatestPackageDependency().Version #>" PrivateAssets="contentfiles;analyzers" />
   </ItemGroup>
 
+  <!-- Would be nice to do this from Xenko targets somehow, but I am unclear how to... -->
+  <Target Name="CopyData" AfterTargets="Publish">
```

Comment:
If this works well, we should move it to `sources\core\Xenko.Core\build\Xenko.Core.targets` so that it doesn't pollute user projects.
Otherwise it will be harder to maintain/update since outside of our control (user can remove/tweak it), and Visual Studio default projects won't work out of the box.

Reply:
Hmm. This comment appears to conflict with your comment above where I tried to add it to the CompilerApp.targets. Maybe I am confused?

Reply:
Sorry for the confusion.
There is two targets to write:
* Copying asset compiler output data (currently called `CopyData`). This should go in `Xenko.Core.Assets.CompilerApp`
* Making .ssdeps (native libs) work. This should go in `Xenko.Core`

Btw, please make both target names starts with either Xenko or _Xenko (depending if public or not, likely _Xenko in your case)

Reply:
After further investigation I can see that this is a custom file format generated from the xenko target file. Will try to figure out how to feed that into a publish copy operation. Any suggestions? Should it be built into an existing C# tool? PackageBuilder for example, new MSBuild Task?

Reply:
Also it feels like we should leverage the built in functionality of nuget/dotnet. Other nuget 3rd party packages accomplish native packaging with no custom code, what is different about Xenko's process?

Reply:
Unfortunately there is no built-in functionality.
The point is not to copy but also to set those libs as native lib.
Each platform **had** a different path for that (`AndroidNativeLibrary` for Android, Copy for UWP but need to choose proper CPU, `NativeReference` + `MtouchExtraArgs` for iOS, etc...).
The idea was to hide those differences so that the csproj just add a single `XenkoNativeLibrary` and it automatically do the proper stuff for each platform.
However, this was done some time ago and it's possible there is easier way to do this (highly doubt it though). Ideally at some point I should write a spec and submit PR for every platform (Xamarin.iOS/Android, dotnet and MSBuild), but I think it would be a lot of work!